src/components/DateSelector/DateSelector.tsx

```diff
@@ -0,0 +1,179 @@
+ * @returns `true` if the string is a valid date in the MM-DD format, otherwise `false`.
+ */
+const isRealDate = (dateString: string): boolean => {
+	if (!isValidDateFormat(dateString)) { return false; }
+
+	const [monthStr, dayStr] = dateString.split('-');
+	const month = Number(monthStr);
+	const day = Number(dayStr);
+
+	const convertedToDate = new Date(2000, month - 1, day);
+	return convertedToDate.getMonth() + 1 === month && convertedToDate.getDate() === day;
+};
+
+const getDaysInMonth = (month: string) => {
+	switch (month) {
+		case '02':
+			return 29; // Always 29 for February
+		case '04':
+		case '06':
+		case '09':
+		case '11':
+			return 30;
+		default:
+			return 31;
+	}
+};
+
+const DateSelector = ({ setDateValue, labelContent = 'Select Date' }: Props) => {
+	const [tempDateValue, setTempDateValue] = useState<string>('');
+	const [month, day] = tempDateValue ? tempDateValue.split('-') : ['', ''];
+
+	const daysInMonth = month ? getDaysInMonth(month) : 31;
+
+	/**
+	 * Generates an array of `<option>` elements representing each day of the month.
+	 *
+	 * @param {number} daysInMonth - The total number of days in the current month.
+	 * @returns {JSX.Element[]} An array of JSX <option> elements for use in a day dropdown.
+	 *
+	 * Each day is zero-padded (e.g., "01", "02", ...) for use as the value attribute,
+	 * while the visible text shows the numeric day (e.g., 1, 2, ...).
+	 */
+	const dayOptions = Array.from({ length: daysInMonth }, (_, i) => {
+		const dayVal = String(i + 1).padStart(2, '0');
+		return (
+			<option key={dayVal} value={dayVal}>
+				{i + 1}
+			</option>
+		);
+	});
+
+	/**
+	 * Handles changes to the date selection inputs (month or day).
+	 *
+	 * When the month is changed, it verifies if the previously selected day is still valid
+	 * for the new month. If the day exceeds the number of days in the selected month,
+	 * the day is cleared and the parent date value is reset to avoid holding an invalid date.
+	 *
+	 * When the day is changed, it simply updates the temporary date state accordingly.
+	 *
+	 * @param {ChangeEvent<HTMLSelectElement>} e - The change event from a `<select>` input (month or day).
+	 */
+	const handleDateInputChange = (e: ChangeEvent<HTMLSelectElement>) => {
+		const { name, value } = e.target;
+
+		setTempDateValue((prev) => {
+			let [prevMonth = '', prevDay = ''] = prev.split('-');
+
+			if (name === 'month') {
+				const newDaysInMonth = getDaysInMonth(value);
+				const newDay = Number(prevDay) > newDaysInMonth ? '' : prevDay;
+
+				if (Number(prevDay) > newDaysInMonth) {
+					setDateValue(''); // Reset parent when day becomes invalid
+				}
+
+				return `${value}-${newDay}`;
+			}
+
+			if (name === 'day') {
+				return `${prevMonth}-${value}`;
+			}
+
+			return prev;
+		});
+	};
+
+	useEffect(() => {
+		if (!isRealDate(tempDateValue)) {
+			setDateValue('');
+			return;
+		}
+		if (isRealDate(tempDateValue)) {
+			setDateValue(tempDateValue);
+		}
+	}, [tempDateValue]);
```

Comment:
This `useEffect` is not needed. When the component prop changes it will get re-rendered anyways. We should instead use a function that receives the prop as input and does this validation.

_If, and only if_, the computation is expensive use a `useMemo` instead.

Reply:
We should add both: A message when the date is invalid, and then a check when submitting to block subissions on invalid date